### PR TITLE
NER Tagging Module (BioNLP through HuggingFace)

### PR DESCRIPTION
**Pull request by mcannon068nw**
Modularization of NER tagging using fine-tuned BioNLP implementation from Huggingface: https://huggingface.co/alvaroalon2

Intended use is to perform NER for a body of text or corpus and attempt to normalize NER-tagged entities to a concept using the VICC normalizers for downstream applications. Single bodies of text can be processed using the *process_text()* method or a list of texts can be tagged using the *batch()* method.

Test examples for trying code are provided:
[_bionlp_example.txt](https://github.com/user-attachments/files/16363588/_bionlp_example.txt)

These can be run by:
```
import bionlp
with open('_bionlp_example.txt') as f:
    lines = f.readlines()
    lines = [line.rstrip('\n') for line in lines]
lines
df = bionlp.batch(lines)
df[0:5]
```

### CODE REVIEW

File: bionlp.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        if row['entity_group'] == 'CHEMICAL':
+            norm_result = __normalize_therapy(row['word'])
+        if row['entity_group'] == 'DISEASE':
+            norm_result = __normalize_disease(row['word'])
+        result.at[index, 'concept_match_type'] = norm_result[0]
+        result.at[index, 'concept_id'] = norm_result[1]
+        result.at[index, 'concept_label'] = norm_result[2]
+    return result
+
+def __normalize_gene(word):
```
Comment (jsstevenson) [inline bionlp.py:44]: single underscore for private functions

File: bionlp.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    r = requests.get(f'https://normalize.cancervariants.org/gene/normalize?q={word}')
```
Comment (jsstevenson) [inline bionlp.py:45]: fair warning, this API is gonna change soonish

**Review by jsstevenson — CHANGES_REQUESTED** [review]

**Review by jsstevenson — APPROVED** [review]

File: bionlp.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    return df
+
+def __tag_chemicals(text):
+    __pipe_chemical = pipeline("token-classification", model="alvaroalon2/biobert_chemical_ner", aggregation_strategy="first")
+    chem_results = __pipe_chemical(text)
+    df = __drop_unknowns(pd.DataFrame(chem_results))
+    df['original_text'] = text
+    return df
+
+def __tag_diseases(text):
```
Comment (jsstevenson) [inline bionlp.py:110]: remember to cut these down to one underscore

**Review by jsstevenson — CHANGES_REQUESTED** [review]

File: bionlp.py (hunk 1 above)
Comment (mcannon068nw) [reply bionlp.py]: change made

File: bionlp.py (hunk 2 above)
Comment (mcannon068nw) [reply bionlp.py:47]: 🫡

File: bionlp.py (hunk 3 above)
Comment (mcannon068nw) [reply bionlp.py]: change made

**Review by jsstevenson — APPROVED** [review]

**Pull request merged**